contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -3991,10 +4001,12 @@ load_categories_hash(RawStmt *cats_sql, MemoryContext per_query_ctx)
  * create and populate the bbf_pivot tuplestore
  */
 static Tuplestorestate *
-get_bbf_pivot_tuplestore(RawStmt *sql,
-						HTAB *bbf_pivot_hash,
-						TupleDesc tupdesc,
-						bool randomAccess)
+get_bbf_pivot_tuplestore(RawStmt 	*sql,
+						const char 	*sourcetext,
+						const char	*funcName,
+						HTAB 		*bbf_pivot_hash,
+						TupleDesc 	tupdesc,
+						bool 		randomAccess)
```

Comment:
What does randomAccess do in this function ?

Reply:
randomAccess: if true, both forward and backward accesses to the tuple store are allowed.

SFRM_Materialize mode expects us to return a NULL Datum. The actual tuples are in our tuplestore and passed back through rsinfo->setResult. rsinfo->setDesc is set to the tuple description that we actually used to build our tuples with, so the caller can verify we did what it was expecting.